### DISCUSSION TITLE
How to update validation error when datatype changed. Currently the error msg will only be updated after the LSP restarted.

### DISCUSSION THREAD

**Discussion by durianwaffle** [eclipse-langium/langium#1724] [Q&A] [answered]
Hi Langium, I implemented validation error for undefined variable.
But how to not be showing the validation error for undefined variable once user created the variable?
Currently, the error msg will only be updated after the LSP restarted.

**Comment by cdietrich** [comment, marked as answer]
Hi, what exactly did you implement. Usually this works ootb.

**Reply by durianwaffle** [reply]
I found the issue. I did a caching for types, so the type is not the same as in the backend.
